mmv1/provider/core.rb

```diff
@@ -190,12 +190,70 @@ def compile_file_list(output_folder, files, file_template, pwd = Dir.pwd)
           Google::LOGGER.debug "Compiling #{source} => #{target}"
           file_template.generate(pwd, source, target, self)
 
+          add_hashicorp_copyright_header(output_folder, target)
           replace_import_path(output_folder, target)
         end
       end.map(&:join)
       Dir.chdir pwd
     end
 
+    def add_hashicorp_copyright_header(output_folder, target)
+      # only add copyright headers when generating TPG and TPGB
+      return unless output_folder.end_with?('terraform-provider-google') ||
+                    output_folder.end_with?('terraform-provider-google-beta')
```

Comment:
Is it safe to assume that the output folder will match the repo name? Is there a better way to achieve this?

Reply:
This makes sense to me! We could possibly use `git` commands but that adds probably needless complexity.

Maybe add a control list of known directory names for outputs and log @ info that "Couldn't identify target folder name {{name}}: output may differ from what you expect." or something.

Reply:
That's a good idea, I'll get on that soon.

Though I then don't want to miss out any repo names, are there ones missing from/incorrect in this list?
- terraform-provider-google
- terraform-provider-google-beta
- terraform-next
- terraform-google-conversion

Reply:
SGTM. This is just gonna be an INFO log I'd assume, so folks can figure out adding their target if we're off. It's more for one-time contributors with weird setups, that flags that something's off.

Reply:
I added some code in [e9d1fc7](https://github.com/GoogleCloudPlatform/magic-modules/pull/7937/commits/e9d1fc7bc158d53ccd2a0c7265ac7d3b07d45635) to address this comment.

Currently I think the info log will only help if a user has cloned the TPG or TPGB repo and renamed it to something different, e.g. `terraform-provider-google-my-local-copy`. I think the info log will help someone in that scenario as they'll need help understanding why there are lots of diffs to _remove_ copyright headers all of a sudden.

If a user cloned terraform-google-conversion and renamed it to terraform-provider-google(or -beta) then the info log won't help them, but I don't know how to handle that unless using git info like you suggested.

Reply:
Yea, that's reasonable- a user who might get surprised is much more likely to have given a repo an arbitrary different name than its base one rather than a wrong name.